packages/shared/src/types.ts

```diff
@@ -98,27 +103,73 @@ export type ContentDisposition = (typeof ValidContentDispositions)[number];
 export const ValidACLs = ["public-read", "private"] as const;
 export type ACL = (typeof ValidACLs)[number];
 
-type RouteConfig = {
+type ImageProperties = {
+  /** Specify the width of the image. */
+  width?: number;
+  /** Specify the height of the image. */
+  height?: number;
+  /**
+   * Specify the aspect ratio of the image.
+   * @remarks If both width and height are specified, this will be ignored.
+   */
+  aspectRatio?: number;
+};
+
+type AdditionalProperties<T> = Record<string, unknown> & T;
+
+type RouteConfig<TAdditionalProperties extends Record<string, unknown>> = {
```

Comment:
Sidequest I remembered when working here - It'd be cool to have more jsdoc driven API reference documentation. Jsdoc should be the driver for the api reference documentation and we shouldnt have to copy-paste stuff in multiple places